scripts/sanitise-db.sql

```diff
@@ -0,0 +1,366 @@
+                'short_desc', substring(lorem_text from 10 for 16),
+                'mitigations', substring(lorem_text from 1 for 10),
+                'consequences', substring(lorem_text from 1 for 5),
+                'risk_owner_role', substring(lorem_text from 1 for 3)
+            )
+        )
+        WHERE data_blob IS NOT NULL;
+
+        GET DIAGNOSTICS row_count = ROW_COUNT;
+        RAISE NOTICE 'UPDATED % ROWS
+        ', row_count;
+    EXCEPTION WHEN unique_violation THEN
+        RAISE NOTICE 'Unique violation error occurred during risk_register detailed descriptions update. Skipping conflicting values.';
+    END;
+END;
+$$ LANGUAGE plpgsql;
+
+
+
+CREATE OR REPLACE FUNCTION anonymize_place_detail_table() RETURNS VOID AS $$
+DECLARE
+    row_count INTEGER;
+    pre_update_count INTEGER;
+BEGIN
+    RAISE NOTICE 'PLACE_DETAIL table: anonymising Name, Email and Telephone keys with fake data in DATA_BLOB column';
+
+    SELECT COUNT(*) INTO pre_update_count FROM place_detail WHERE data_blob IS NOT NULL;
+    RAISE NOTICE 'Number of rows to be checked: %', pre_update_count;
+
+    BEGIN
+        UPDATE place_detail
+        SET data_blob = jsonb_set(
+            data_blob,
+            '{answer}',
+            COALESCE(
+                CASE
+                    WHEN data_blob->>'indicator' = 'Name' THEN to_jsonb(anon.fake_last_name() || ' ' || anon.fake_last_name())
+                    WHEN data_blob->>'indicator' IN ('Email', 'Contact email') THEN to_jsonb(anon.fake_email())
+                    WHEN data_blob->>'indicator' = 'Telephone' THEN to_jsonb(anon.random_phone('+44'))
+                    ELSE data_blob->'answer'
+                END,
+                data_blob->'answer'
+            )
+        )
+        WHERE data_blob IS NOT NULL
+        AND (data_blob->>'indicator' IN ('Name', 'Email', 'Contact email', 'Telephone'));
+
+        GET DIAGNOSTICS row_count = ROW_COUNT;
+        RAISE NOTICE 'UPDATED % ROWS', row_count;
+    EXCEPTION WHEN unique_violation THEN
+        RAISE NOTICE 'Unique violation error occurred during place_detail update. Skipping conflicting values.';
+    END;
+END;
+$$ LANGUAGE plpgsql;
+
+
+
+
+
+
+DO $$
+BEGIN
+    PERFORM anonymize_with_lorem_ipsum('funding'::TEXT, 'data_blob'::TEXT, 'funding_source'::TEXT);   --- LARGE TABLE
+    PERFORM anonymize_with_lorem_ipsum('funding_comment'::TEXT, 'data_blob'::TEXT, 'comment'::TEXT);
+    PERFORM anonymize_with_lorem_ipsum('private_investment'::TEXT, 'data_blob'::TEXT, 'additional_comments'::TEXT);
+    PERFORM anonymize_with_lorem_ipsum('programme_progress'::TEXT, 'data_blob'::TEXT, 'answer'::TEXT);
+
+    PERFORM anonymize_postcode_in_column('project_dim', 'postcodes');
+
+    PERFORM anonymize_postcode_in_data_blob('project_dim', 'locations');
+
+    PERFORM anonymize_region('programme_dim', 'programme_name');
+    PERFORM anonymize_region('organisation_dim', 'organisation_name');
+    PERFORM anonymize_region('project_dim', 'project_name');
```

Comment:
Do we need to anonymise these? I feel like they're all public information (and they make the Find view look very weird

<img width="834" alt="image" src="https://github.com/user-attachments/assets/39adf6b2-819f-4fa6-8506-34f65be4509e">

Reply:
I no longer _need_ these to be de-anonymised for my e2e tests, but still do consider if it's correct to anonymise this field or not

Reply:
Yeah it doesn't look nice we prob need to re-think about it, if we want to anonymise this or not then we can change accordingly.

Reply:
Agree, happy not to anonymise the organisations, but we probably do want to anonymise any free text about the projects.

Reply:
Lorem_ipsum is covering the free text. Should we remove anon on project_name, organisation_name and programme_name?